server/api/controllers/class/index.js

```diff
@@ -2,46 +2,79 @@ const ModelsRoot = require('../../models');
 
 let Models = null;
 let db = null;
-let container = {};
+let classContainer = {};
 
-container.readOne = async (id) => {
+// TODO:PUSH:
+classContainer.readOne = async (id) => {
   try {
-    return await Models.class.findOne({
+    let record = await Models.class.findOne({
       where: { id: id }
     });
+
+    if (record instanceof Error) throw record;
+
+    return record;
   } catch (error) {
     return error;
   }
 };
 
-container.writeOne = async (data) => {
+// TODO:PUSH:
+classContainer.writeOne = async (data) => {
   try {
-    return await Models.class.create(data);
+    let record = await Models.class.create(data);
+
+    if (record instanceof Error) throw new Error('Record must be complete');
+
+    return record;
   } catch (error) {
     return error;
   }
 };
 
-container.updateOne = async (data) => {
+// TODO:PUSH:
+classContainer.updateOne = async (data, id, method) => {
   try {
-    return await Models.class.update(data, {
-      where: { id: data.id }
+    let record = await classContainer.readOne(id);
+
+    if (method === 'PUT') {
+      if (!record) {
+        record = await classContainer.writeOne(data);
+
+        return record;
+      }
+
+      let containsAllKeys = Object.keys(record.dataValues).every((recKey) => {
+        return Object.keys(data).includes(recKey);
+      });
+
```

Comment:
This method will throw a fail Error if the request was `PUT` because a `PUT` must provide a complete data set for the Database in order to update the record.  The `PATCH` will handle only sections of the record to be updated.  Currently there are really only two values to be updated: name and senior.  Senior is a boolean indicating if they are currently seniors.  

Some validation needs to be configured to ensure only one class can be a senior and if a class is moved to senior status, the previous Seniors will be set to non seniors.